sys/arduino/include/spiport.hpp

```diff
@@ -0,0 +1,233 @@
+    /**
+     * @brief   Transfer two bytes of data
+     * @param[in]       data        The two bytes to send
+     * @return  The received two bytes
+     *
+     * Arduino is sending the most significant byte first, if the SPI interface
+     * is configured to send the most significant bit first. If the least
+     * significant bit is send first, Arduino will also send the least
+     * significant byte first.
+     *
+     * This wrapper currently only supports sending the most significant bit
+     * first over the wire, so this function will also always send the most
+     * significant byte first.
+     */
+    uint16_t transfer16(uint16_t data)
+    {
+        data = htons(data);
+        transfer(&data, sizeof(data));
+        return ntohs(data);
+    }
+
+    /**
+     * @brief   Transfer data
+     * @param[inout]    buf     Buffer containing the data to send, received
+     *                          data will be written here
+     * @param[in]       count   Number of bytes to send
+     */
+    void transfer(void *buf, size_t count);
+
+    /**
+     * @brief   Doesn't do anything, for compatibility only
+     */
+    void begin() { }
+
+    /**
+     * @brief   Doesn't do anything, for compatibility only
+     */
+    void end() { }
+
+    /**
+     * @brief   Acquires the SPI interface and applies the given settings
+     * @param[in]   settings    Settings to apply
+     */
+    void beginTransaction(SPISettings settings);
+
+    /**
+     * @brief   Releases the SPI interface
+     */
+    void endTransaction();
+
+    /**
+     * @brief   Sets the bit order to the given value
+     *
+     * @details Currently only most significant bit first is supported, as
+     *          practically no hardware exists using lsbfirst. An assertion
+     *          is triggered if lsbfirst is requested.
+     * @deprecated  This function is deprecated in the official Arduino API,
+     *              so it is a good idea to not use it. In RIOT, this function
+     *              is not yet scheduled for removal to allow using Arduino
+     *              libraries using it.
+     */
+    void setBitOrder(uint8_t order);
+
+    /**
+     * @brief   Sets the SPI mode (clock phase and polarity)
+     *
+     * @deprecated  This function is deprecated in the official Arduino API,
+     *              so it is a good idea to not use it. In RIOT, this function
+     *              is not yet scheduled for removal to allow using Arduino
+     *              libraries using it.
+     */
+    void setDataMode(uint8_t mode);
+
+    /**
+     * @brief   Sets the SPI clock in an archaic manner
+     *
+     * @deprecated  This function is deprecated in the official Arduino API,
+     *              so it is a good idea to not use it. In RIOT, this function
+     *              is not yet scheduled for removal to allow using Arduino
+     *              libraries using it.
+     */
+    void setClockDivider(uint8_t divider);
+};
+
+extern SPIClass SPI;
```

Comment:
Please add the following to make `doxygen` and Travis happy.

```suggestion
/**
 * @brief: Instance of the SPI interface as required by the Arduino API
 */
extern SPIClass SPI;
```

Reply:
Thanks for pointing this out. Done! May I squash?

Reply:
Yes, please.